src/lib/nodemailer.ts

```diff
@@ -1,35 +1,83 @@
+// import prisma from '../db/client'
+// import { PrismaClient } from '@prisma/client';
 import nodemailer from 'nodemailer'
+import prisma from './db/client';
 
-const transporter = nodemailer.createTransport({
-  host: process.env.NODEMAILER_TRANSPORTER_HOST,
-  port: parseInt(process.env.NODEMAILER_TRANSPORTER_PORT!),
-  secure: process.env.NODEMAILER_TRANSPORTER_SECURE === 'true',
-  auth: {
-    user: process.env.NODEMAILER_USERNAME,
-    pass: process.env.NODEMAILER_PASSWORD,
-  },
-})
-
-export const sendPassRecoveryMail = async ({recipient, token}: any) => {
+// const prisma = new PrismaClient();
+
+let smtpDetails: any;
+
+async function getSmtpDetails() {
+  try {
+    smtpDetails = await prisma.mailSetting.findFirst();
+    return smtpDetails;
+  } catch (error) {
+    console.error('Error retrieving SMTP details:', error);
+    throw error;
+  } finally {
+    await prisma.$disconnect();
+  }
+}
+
+async function generateTransporter() {
+  const smtpDetails = await getSmtpDetails();
+
+  if (smtpDetails) {
+    const transporter = nodemailer.createTransport({
+      debug: true,
+      logger: true,
+      host: smtpDetails.host,
+      port: smtpDetails.port,
+      secure: smtpDetails.secure,
+      auth: {
+        user: smtpDetails.username,
+        pass: smtpDetails.password,
+      }
+    });
+
+    // verify connection configuration
+    transporter.verify(function (error) {
+      if (error) {
+        console.log(error);
+      } else {
+        console.log("Server is ready to take our messages");
+      }
+    });
+
+    return transporter;
+  } else {
+    console.error('SMTP not found');
+    return null;
+  }
+}
+
+export const sendPassRecoveryMail = async ({ recipient, token }: any) => {
   const link = `${process.env.NEXTAUTH_URL}/recover-password?token=${token}`
   const html = `
-      <p>You have requasted an password reovery. Follow the link below to recover your password</p>
+      <p>You have requested ach password recovery. Follow the link below to recover your password</p>
       <div>
         <a href="${link}">Click here</a> to recover your password.
       </div>
       <p>Password recovery session will be closed in 5 minutes.</p>
   `
 
   const message = {
-    from: process.env.NODEMAILER_USERNAME,
+    from: "support@notification.codetisan.com",
```

Comment:
Add a new field in the SMTP settings to let users add "From email" through the form so that it's configurable.